### ISSUE TITLE
Feature Request - 2FA Notification script

### ISSUE THREAD

**Issue by boredazfcuk** [ndbroadbent/icloud_photos_downloader#105]
Hi,
Would it be possible to get an additional command line option to notify when 2FA has expired via an external script? e.g. --notification-script /path/to/my/script/notify.sh

I currently use Prowl for iOS push notifications so don't have e-mail configured for external relay. My script would just contain:

`curl https://api.prowlapp.com/publicapi/add -F apikey=$APIKEY -F application="icloud_photos_downloader" -F event="2FA Expired" -F priority="2" -F description="Two factor authentication has expired. Please renew."
`

This method could also be used with other push notification services such as Pushover, PushBullet, NotifyMyAndroid etc

Thanks.

**Comment by ndbroadbent** [comment]
Hey, that's a great idea! I like `--notification-script`. Would you be able to submit a PR with this change? I think you'll just need to add a `@click.argument` to `icloudpd/base.py`, and then add a `notification_script` argument at the bottom of `def main(`. Then you'll need to add:

```pytho
import subprocess

# ...

subprocess.call([notification_script])
```

Hope that helps!

**Comment by boredazfcuk** [comment]
I think you've over-estimated my Python skills, which I can sum up in two words... 'hello world!' lol

I should be able to work it out, it may take a while though. I'll see what I can come up with this afternoon and report back.

**Comment by boredazfcuk** [comment]
Well I've got something that's working on my system...
https://github.com/boredazfcuk/icloud_photos_downloader/blob/master/icloudpd/base.py

Does that look right to you?

**Comment by ndbroadbent** [comment]
Awesome thanks testing that out! I merged those changes in and added a test for it. Released version `1.3.3` that includes the `--notification-script` option
